s3po/backends/s3.py

```diff
@@ -93,3 +93,13 @@ def func():
             multi.upload_part_from_file(StringIO(data), count)
             multi.complete_upload()
             return True
+
+    def list(self, bucket, prefix=None, delimiter=None, headers=None):
+        '''List the bucket, possibly limiting the search with a prefix.'''
+        bucket = self.conn.get_bucket(bucket)
+        if prefix is None:
+            prefix = ''
+        if delimiter is None:
+            delimiter = ''
+        # Consumer iterator to make a list to keep parity with Swift backend
```

Comment:
Also, I think we should feel comfortable returning an iterable where possible. If the backend doesn't provide it, don't worry, but there's no reason to aggressively sacrifice that functionality.